src/Hangfire.SqlServer/SqlServerConnection.cs

```diff
@@ -341,6 +341,26 @@ public override int RemoveTimedOutServers(TimeSpan timeOut)
                 new { timeOutAt = DateTime.UtcNow.Add(timeOut.Negate()) }));
         }
 
+        public override List<string> FetchAndRemoveTimedOutServers(TimeSpan timeOut)
+        {
+            if (timeOut.Duration() != timeOut)
+            {
+                throw new ArgumentException("The `timeOut` value must be positive.", "timeOut");
+            }
+
+            string query = $@"select Id from [{nameof(_storage)}].Server where LastHeartbeat < @timeOutAt";
+
+            var servers = _storage.UseConnection(connection => connection
+                .Query<string>(query, new { timeOutAt = DateTime.UtcNow.Add(timeOut.Negate()) }))
+                .ToList();
+
+            _storage.UseConnection(connection => connection.Execute(
```

Comment:
sorry my bad i miss read the code i thought the code was doing similar things you can remove this